Add sensor support: Honeywell ABP (SPI version)

RubyBailey opened the pull request:
# What does this implement/fix? 

Adding Honeywell ABP sensor (SPI version). I have used this sensor for a number of personal applications. I have appreciated the support from the esphome community in the past and wanted to contribute to esphome.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

None

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#1880

In progress

## Test Environment

- [x] ESP32
- [ ] ESP32 IDF
- [x] ESP8266

## Example entry for `config.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

```yaml
# Example config.yaml
sensor:
- platform: honeywellabp
    pressure:
        name: "Honeywell pressure"
        unit_of_measurement: "psi"
        min_pressure: 0
        max_pressure: 15
    temperature:
        name: "Honeywell temperature"
    cs_pin: GPIO5 #D5
    update_interval: 2s
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).
  
If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

RubyBailey committed:
Merge pull request #1 from esphome/dev

Merging from esphome

RubyBailey committed:
Merge branch 'esphome:dev' into dev

RubyBailey committed:
Merge branch 'esphome:dev' into dev

RubyBailey committed:
Merge branch 'esphome:dev' into dev

RubyBailey committed:
initial commit

Commit:
added codeowners

Commit:
updated codeowners

Commit:
updated codeowners

Commit:
updated

Commit:
updated

Commit:
update

Commit:
updated

Commit:
updated

Commit:
updated

Commit:
updated

Commit:
updated

Commit:
updated

Commit:
updated

Commit:
updated

RubyBailey commented:
Checks are unsuccessful because it wants me to "Please run `script/build_codeowners.py`" 
This is likely a simple request for many of you but I have no idea how to do that. If I try python or python3 script/build_codeowners.py I just get error messages. 
I have added myself to the CODEOWNERS file, but the check doesn't accept that that.
Would someone be able to help me get past this hurdle. Thanks in advance.

OttoWinter commented on esphome/components/honeywellabp/honeywellabp.h:
```diff
@@ -0,0 +1,46 @@
+namespace esphome {
+namespace honeywellabp {
+
+class HONEYWELLABPSensor : public sensor::Sensor,
```
Why does this inherit from Sensor? It doesn't publish anything itself (only via the subsensors pressure and temperature). You can remove this.

RubyBailey replied:
Thanks for the comment. I have updated the code.

OttoWinter commented on esphome/components/honeywellabp/sensor.py:
```diff
@@ -0,0 +1,68 @@
+            cv.Optional(CONF_PRESSURE): sensor.sensor_schema(
+                accuracy_decimals=1,
+                device_class=DEVICE_CLASS_PRESSURE,
+                state_class=STATE_CLASS_MEASUREMENT,
```
Does this sensor not have a unit?

RubyBailey replied:
My bad; I could not find a UNIT_PSI in consp.py so I decided not to add any unit_of_measurement. I have now changed that in the code to default to psi. Default also updated in documentation PR.

Commit:
updated

Commit:
format updated

Commit:
added codeowner

Commit:
updated

Commit:
updated codeowners

RubyBailey commented:
I figured out how to run the script/codeowners.py. Updated __inti__.py with CODEOWNERS=["@RubyBailey"]. Ran script and it updated the CODEOWNERS file. Deleted CODEOWNERS=["@RubyBailey"] from __init__.py. Uploaded CODEOWNERS file to this PR yet check still fails.

Anyone know why?

Commit:
updated  codeowners (test)

Commit:
Updated CODEOWNERS

RubyBailey commented:
@Naesstrom I noticed that you requested the integration of this sensor into esphome (esphome / feature-request #475). Your request was for a I2C interface; I only have sensors available with SPI interface.
Would you have a SPI version sensor, be able to test run the submitted code and look at the documentation? 
Thanks

Commit:
updated

RubyBailey commented:
@OttoWinter 
I cannot figure out why Lint says that the CODEOWNERS file is not up-to-date. 

Output on my end:
:~/esphome$ script/build_codeowners.py
Wrote CODEOWNERS
:~/esphome$ script/ci-custom.py
:~/esphome$ script/build_codeowners.py --check
CODEOWNERS file is up to date

Yet CODEOWNERS file when uploaded still fails Lint. 
Any advice would be appreciated. Thanks much.

OttoWinter commented on esphome/components/honeywellabp/sensor.py:
```diff
@@ -0,0 +1,69 @@
+)
+
+DEPENDENCIES = ["spi"]
+
```
```suggestion
CODEOWNERS = ["@RubyBailey"]

```

RubyBailey replied:
Thanks, much appreciated.

RubyBailey replied:
From:
https://esphome.io/guides/contributing.html

5. Extras
"Note : This serves as documentation for some of ESPHome’s internals and is not necessarily part of the development guide."

"CODEOWNERS: GitHub usernames or team names of people that are responsible for this integration. You should add at least your GitHub username here, as well as anyone who helped you to write code that is being included."

It wasn't clear to me that CODEOWNERS was required in a python file. It may be beneficial to emphasis that it must be added to the python file for acceptance of the code by the automatic tools.

jesserockz replied:
@RubyBailey That `CODEOWNERS` item is in a list for python variables =)

```
All Python modules have some magic symbols that will automatically be loaded by the ESPHome loader. These are:
```

RubyBailey replied:
Yes I know. I was just pointing out that the wording wasn't clear to me that it was required and may not be clear to others. No need to belabour the point any longer ☺. Thanks to all of you who have helped me with esphome issues. Hope this component helps someone else. Again, thanks.

OttoWinter reviewed (approved):
Code LGTM now, thanks! (only codeowners needs to be updated, see comment)

OttoWinter commented:
Please also add a basic entry in one of the yaml files in the `tests/` folder, this is only there to test it compiles correctly with the output the codegen creates.

Commit:
added honetwellabp
